.setup/INSTALL_SUBMITTY_HELPER.sh

```diff
@@ -669,6 +669,32 @@ popd > /dev/null
 chown -R ${DAEMON_USER}:${COURSE_BUILDERS_GROUP} ${SUBMITTY_INSTALL_DIR}/SubmittyAnalysisTools
 chmod -R 555 ${SUBMITTY_INSTALL_DIR}/SubmittyAnalysisTools
 
+################################################################################################################
+################################################################################################################
+# BUILD AND INSTALL ANALYSIS TOOLS TS
+
+echo -e "Build and install analysis tools ts"
+
+mkdir -p "${SUBMITTY_INSTALL_DIR}/SubmittyAnalysisToolsTS"
+
+ANALYSIS_TOOLS_TS_REPO="${SUBMITTY_INSTALL_DIR}/GIT_CHECKOUT/AnalysisToolsTS/"
+
+# Copy cloned files to AnalysisToolsTS directory
+rsync -rtz "${ANALYSIS_TOOLS_TS_REPO}" "${SUBMITTY_INSTALL_DIR}/SubmittyAnalysisToolsTS"
+
+pushd "${SUBMITTY_INSTALL_DIR}/SubmittyAnalysisToolsTS"
+
+# # install npm packages
+sudo npm install
+
+# # build project
+sudo npm run build
```

Comment:
In general, we should probably not be using `sudo` when installing dependencies since doing so allows the dependencies we're installing to have a lot of unnecessary permissions (including, in the extreme case, potentially deleting all the files on the system).  I think this could be resolved by changing the permissions on the relevant files but I'm not sure what the correct permissions should be.  Perhaps @bmcutler can advise here?

Reply:
Update to run the `npm install` from `daemon user`. Gave daemon user `rwx `permissions to the directory `SubmittyAnalysisToolsTS`.  rwx permissions are needed due to creation of new directories in `npm install`.